packages/mobx-state-tree/src/middlewares/on-action.ts

```diff
@@ -220,18 +220,24 @@ export function recordActions(
  *
  * @param target
  * @param listener
- * @param attachAfter (default false) fires the listener *after* the action has executed instead of before.
+ * @param options boolean (legacy attachAfter, default false) or object that controls the behavior
+ * @param options.attachAfter (default false) fires the listener *after* the action has executed instead of before.
+ * @param options.onlyOuter (default true) only fires the listener for outermost actions
  * @returns
  */
+export interface IOnActionOptions { attachAfter?: boolean, onlyOuter?: boolean }
 export function onAction(
     target: IAnyStateTreeNode,
     listener: (call: ISerializedActionCall) => void,
-    attachAfter = false
+    attachAfterOrOptions: boolean | IOnActionOptions = false
 ): IDisposer {
     // check all arguments
+    const { attachAfter = false, onlyOuter = true } = typeof attachAfterOrOptions === "object"
```

Comment:
Default true properties like this can lead to hard to find issues. I pointed that out here: https://concord-consortium.slack.com/archives/C0303QUNU/p1664938148115039

Maybe the property could be `allActions` with a default of false?